rules/DeadCode/Rector/Array_/RemoveDuplicatedArrayKeyRector.php

```diff
@@ -94,6 +96,14 @@ private function resolveDuplicateKeysArrayItems(Array_ $array): array
                 continue;
             }
 
+            if ($arrayItem->key instanceof CallLike) {
+                continue;
+            }
+
+            if ($arrayItem->key instanceof PropertyFetch) {
+                continue;
+            }
```

Comment:
this should use `ExprAnalyzer->isDynamicExpr()`

Reply:
I created PR for it, with additional fixture test:

- https://github.com/rectorphp/rector-src/pull/7000